go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -454,6 +454,19 @@ func (bnc *BaseNetworkController) addLogicalPortToNetwork(pod *kapi.Pod, nadName
 		return nil, nil, nil, false, fmt.Errorf("[%s] failed geting expected switch name when adding logical switch port: %v", podDesc, err)
 	}
 
+	if bnc.TopologyType() == ovntypes.LocalnetTopology {
```

Comment:
Can we move this to the localnet controller ?

Reply:
looking on `go-controller/pkg/ovn/secondary_localnet_network_controller.go` 
i dont see where it fit,
if you still think it should be there, can you suggest where please?

Reply:
So since this is a check that should prevent localnet network infra to be created we can put it directly at the localne event handler add, update and sync functions at https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/go-controller/pkg/ovn/secondary_localnet_network_controller.go#L96-L124

Just wrap the check with all the logic at a helper with the event handler as receiver and call it from the three places, before calling common network functions.

Reply:
this operation is expansive, so i dont think we need to have it on sync
also having it on 3 places makes it harder to maintain + the current file already has switch for the type
lets please see what @maiqueb  think

Reply:
I think @qinqon 's right. 

While this is less code, it's polluting the [base_network_controller_pods.go](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5000/files#diff-941b484ab7f41b84a6ba9c2dee91f3454401447db0ca70cb401a8cc582cc072e) with something localnet specific ...

Let's at least try to be good citizens. If it proves to be too complex, let's leave it as is now. 
But you have more than enough time to provide nice code.

Reply:
can this be done please on a follow-up ?
it will make it easier to be focused

Reply:
@oshoval this do not merit a follow up, just make a helper function with localnet controller as receiver put the logic there, then you call this a proper places at localnet controller, we have already a lot of spaghetti code at base controller.